launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -239,6 +243,52 @@ QSet<QString> MinecraftInstance::traits() const
     return profile->getTraits();
 }
 
+void MinecraftInstance::populateLaunchMenu(QMenu* menu)
+{
+    QAction* normalLaunch = menu->addAction(tr("&Launch"));
+    normalLaunch->setShortcut(QKeySequence::Open);
+    QAction* normalLaunchOffline = menu->addAction(tr("Launch &Offline"));
+    normalLaunchOffline->setShortcut(QKeySequence(tr("Ctrl+Shift+O")));
+    QAction* normalLaunchDemo = menu->addAction(tr("Launch &Demo"));
+    normalLaunchDemo->setShortcut(QKeySequence(tr("Ctrl+Alt+O")));
+
+    normalLaunchDemo->setEnabled(supportsDemo());
+
+    connect(normalLaunch, &QAction::triggered, [this] { APPLICATION->launch(shared_from_this()); });
+    connect(normalLaunchOffline, &QAction::triggered, [this] { APPLICATION->launch(shared_from_this(), false, false); });
+    connect(normalLaunchDemo, &QAction::triggered, [this] { APPLICATION->launch(shared_from_this(), false, true); });
+
+    QString profilersTitle = tr("Profilers");
+    menu->addSeparator()->setText(profilersTitle);
+
+    auto profilers = new QActionGroup(menu);
+    profilers->setExclusive(true);
+    connect(profilers, &QActionGroup::triggered, [this](QAction* action) {
+        settings()->set("Profiler", action->data());
+        emit profilerChanged();
+    });
+
+    QAction* noProfilerAction = menu->addAction(tr("&No Profiler"));
+    noProfilerAction->setData("");
+    noProfilerAction->setCheckable(true);
+    noProfilerAction->setChecked(true);
+    profilers->addAction(noProfilerAction);
+
+    for (auto profiler = APPLICATION->profilers().begin(); profiler != APPLICATION->profilers().end(); profiler++) {
+        QAction* profilerAction = menu->addAction(profiler.value()->name());
+        profilers->addAction(profilerAction);
+        profilerAction->setData(profiler.key());
+        profilerAction->setCheckable(true);
+        profilerAction->setChecked(settings()->get("Profiler").toString() == profiler.key());
+
+        QString error;
+        if (profiler.value()->check(&error))
+            profilerAction->setIcon(APPLICATION->getThemedIcon("status-good"));
+        else
+            profilerAction->setIcon(APPLICATION->getThemedIcon("status-bad"));
```

Comment:
these icons hide the checkbox. I think we should just disable the items if they are not available

Reply:
problem is, a disabled item can be selected which is weird

Reply:
I just tried it locally and it works as expected. I can't click/select the disabled item

Reply:
> I just tried it locally and it works as expected. I can't click/select the disabled item

if the profiler's validity changes

Reply:
kind of a fusion issue. i wonder if there's a workaround.

Reply:
Can we just keep it like this because it's Fusion's fault?

Reply:
> Disabling items leads to confusing UX imo

I think it's much more confusing if you can't tell that something is enabled/selected

Reply:
huh? selecting an invalid profiler just makes a dialog box appear on launch

Reply:
![image](https://github.com/PrismLauncher/PrismLauncher/assets/57493648/67b24ae8-4afd-4975-a78f-4cf31683a892)
would you agree this is fine though?
I suppose i can change it just for the sake of fusion :pensive:

Reply:
Why not just disable the menu items? That's how it already works to develop. Disabled buttons/menu items is a common UI pattern on desktop systems.

> I suppose i can change it just for the sake of fusion

Please don't forget that Fusion *is* the primary style for Prism Launcher.

Reply:
I guess disabling is fine. We can't add a tooltip though so it could be confusing.